### PR TITLE
NMS-16357: Hide credentials in requisition URL

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-16357

### CODE REVIEW

File: opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/AbstractForeignSourceRepository.java [hunk 1]
```diff
@@ -63,13 +63,22 @@ public AbstractForeignSourceRepository() {
...
         requisition.setResource(resource);
         save(requisition);
         return requisition;
     }
 
+    static String stripCredentials(final Object object) {
+        if (object == null) {
+            return null;
+        } else {
+            return String.valueOf(object).replaceAll("(username=)[^;&]*(;&)?", "$1***$2")
```
Comment (mmahacek) [inline opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/AbstractForeignSourceRepository.java:77]: Does this regex catch if the password happens be the last query parameter of the URL?

Comment (RangerRick) [reply opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/AbstractForeignSourceRepository.java:77]: the `?` at the end makes it optional, I think, if I'm reading it right...

**Review by RangerRick — DISMISSED** [review]

File: opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/AbstractForeignSourceRepository.java (hunk 1 above)
Comment (christianpape) [reply opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/AbstractForeignSourceRepository.java:77]: Added test.

Comment (christianpape) [reply opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/AbstractForeignSourceRepository.java:77]: Ok?

**Review by mmahacek — APPROVED** [review]

**Review by RangerRick — APPROVED** [review]

**Pull request merged**